src/main/java/org/prebid/server/auction/mediatypeprocessor/BidderMediaTypeProcessor.java

```diff
@@ -48,21 +48,23 @@ public MediaTypeProcessingResult process(BidRequest bidRequest, String supported
     }
 
     private Set<MediaType> extractSupportedMediaTypes(BidRequest bidRequest, String supportedBidderName) {
-        final BidderInfo.CapabilitiesInfo capabilitiesInfo = bidderCatalog
-                .bidderInfoByName(supportedBidderName).getCapabilities();
+        final BidderInfo.CapabilitiesInfo capabilitiesInfo = bidderCatalog.bidderInfoByName(supportedBidderName)
+                .getCapabilities();
 
-        final List<MediaType> supportedMediaTypes;
+        final Supplier<BidderInfo.PlatformInfo> fetchSupportedMediaTypes;
         if (bidRequest.getSite() != null) {
-            supportedMediaTypes = ObjectUtil.getIfNotNull(
-                    capabilitiesInfo.getSite(), BidderInfo.PlatformInfo::getMediaTypes);
+            fetchSupportedMediaTypes = capabilitiesInfo::getSite;
+        } else if (bidRequest.getApp() != null) {
+            fetchSupportedMediaTypes = capabilitiesInfo::getApp;
         } else {
-            supportedMediaTypes = ObjectUtil.getIfNotNull(
-                    capabilitiesInfo.getApp(), BidderInfo.PlatformInfo::getMediaTypes);
+            fetchSupportedMediaTypes = capabilitiesInfo::getDooh;
         }
 
-        return CollectionUtils.isNotEmpty(supportedMediaTypes)
-                ? EnumSet.copyOf(supportedMediaTypes)
-                : EnumSet.noneOf(MediaType.class);
+        return Optional.ofNullable(fetchSupportedMediaTypes.get())
+                .map(BidderInfo.PlatformInfo::getMediaTypes)
+                .filter(mediaTypes -> !mediaTypes.isEmpty())
+                .map(EnumSet::copyOf)
+                .orElseGet(() -> EnumSet.noneOf(MediaType.class));
```

Comment:
If we will have `EnumSet.noneOf(MediaType.class)` as constant, then orElseGet can be replaced with orElse